### PR TITLE
nixos/stargazer: fix route ordering

### PR DESCRIPTION
**Pull request by gaykitty**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by gaykitty** [comment]
@AndersonTorres Okay, I have modified the route generation to maintain their order. Instead of being a set of sets like:
```
routes = {
  "localhost" = {
    root = "/path";
  };
};
```

It's a list of sets where the key `route` becomes the section name
```
routes = [
  {
    route = "localhost";
    root = "/path";
  }
];
```

File: nixos/modules/services/web-servers/stargazer.nix [hunk 1]
```diff
@@ -21,143 +20,174 @@ let
...
       type = types.bool;
       default = true;
-      description = lib.mdDoc ''
+      description = mdDoc ''
         Set to false to disable automatic certificate generation.
         Use if you want to provide your own certs.
       '';
     };
 
-    regenCerts = lib.mkOption {
```
Comment (AndersonTorres) [inline nixos/modules/services/web-servers/stargazer.nix:107]: Those cosmetic changes, put them in another commit.

Better yet, undo them. `with lib;` is a plague we should remove from the codebase.

**Comment by AndersonTorres** [comment]
Is it working now?

File: nixos/modules/services/web-servers/stargazer.nix (hunk 1 above)
Comment (gaykitty) [reply nixos/modules/services/web-servers/stargazer.nix:107]: Done. I've broken it out into 2 commits and remove `with lib;`.

**Comment by gaykitty** [comment]
> Is it working now?

In my testing it seems to, yes.
